jfix-stdlib-batching/src/main/java/ru/fix/stdlib/batching/BatchingManagerThread.java

```diff
@@ -172,12 +166,9 @@ private boolean waitForAvailableBatchProcessThread() {
         boolean isBatchProcessorThreadAvailable = false;
         while (!isShutdown.get() && !isBatchProcessorThreadAvailable) {
             try {
-                isBatchProcessorThreadAvailable =
-                        batchProcessorsTracker.tryAcquire(100, TimeUnit.MILLISECONDS);
-                if (isBatchProcessorThreadAvailable) {
-                    log.trace("isBatchProcessorThreadAvailable");
-                    batchProcessorsTracker.release();
-                }
+                isBatchProcessorThreadAvailable = batchProcessorsTracker.isBatchProcessorThreadAvailable(() ->
```

Comment:
I don't yet understand why we are moving from semaphore based logic to wait/notify based. We will discuss it